### PR TITLE
Implementation of cross-correlation, CrossCorrelation test class, auto-correlation and AutoCorrelation test class

### PR DESCRIPTION
**Pull request by AthinaSw**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/maths/AutoCorrelation.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                count++;
+            }
+
+
+        }
+
+        //The calculated auto-correlation of x signal is returned here.
+        return result;
+
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/AutoCorrelation.java:80]: Please remove excessive empty lines

File: src/main/java/com/thealgorithms/maths/AutoCorrelation.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     * @param x The discrete signal
+     * @return The result of the auto-correlation of signals x. The result is also a signal.
+     */
+    public static double[] autoCorrelation(double[] x){
+        //The result signal's length is double the length of signal x minus 1
+        double[] result=new double[2*x.length-1];
+        int N=result.length;
+
+        /*
+        To find the auto-correlation of a discrete signal x, we perform cross-correlation between x signal and itself.
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/AutoCorrelation.java:26]: Could you just call `crossCorrelation(x, x)` then?

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/maths/AutoCorrelation.java (hunk 1 above)
Comment (AthinaSw) [reply src/main/java/com/thealgorithms/maths/AutoCorrelation.java:80]: Okay, sure! I will.

File: src/main/java/com/thealgorithms/maths/AutoCorrelation.java (hunk 2 above)
Comment (AthinaSw) [reply src/main/java/com/thealgorithms/maths/AutoCorrelation.java:26]: Ok, I was debating between choosing class autonomy or least amount of code lines. If class autonomy is not neccessary then I will do it this way

Comment (AthinaSw) [reply src/main/java/com/thealgorithms/maths/AutoCorrelation.java:26]: Ok, I was debating between choosing class autonomy or least amount of code lines. If class autonomy is not neccessary than I will do it this way

File: src/test/java/com/thealgorithms/maths/AutoCorrelationTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * @version 1.0
+ */
+public class AutoCorrelationTest {
+
+    @Test
+    void testAutoCorrelation(){
+        double[] x={1,2,1,1};
+        double[] y={1,2,3,4,5};
+        double[] xx={1,3,5,7,5,3,1};
+        double[] yy={5,14,26,40,55,40,26,14,5};
```
Comment (siriak) [inline src/test/java/com/thealgorithms/maths/AutoCorrelationTest.java:21]: Please create a parameterized test

File: src/test/java/com/thealgorithms/maths/CrossCorrelationTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * Test class for CrossCorrelation class
+ *
+ * @author Athina-Frederiki Swinkels
+ * @version 1.0
+ */
+public class CrossCorrelationTest {
+
+    @Test
+    void testCrossCorrelation(){
+        double[] x={1,2,1,1};
```
Comment (siriak) [inline src/test/java/com/thealgorithms/maths/CrossCorrelationTest.java:18]: Please create a parameterized test

**Review by siriak — CHANGES_REQUESTED** [review]

**Comment by siriak** [comment]
Please fix the PR check
image

**Comment by AthinaSw** [comment]
In the details of the "Clang format linter / build" , it says this : 
--- ./src/test/java/com/thealgorithms/maths/AutoCorrelationTest.java	(original)
+++ ./src/test/java/com/thealgorithms/maths/AutoCorrelationTest.java	(reformatted)
@@ -34,4 +34,4 @@
         }
         return result;
     }
-}+}
I don't really understand what I sould do here

**Comment by AthinaSw** [comment]
> In the details of the "Clang format linter / build" , it says this : --- ./src/test/java/com/thealgorithms/maths/AutoCorrelationTest.java (original) +++ ./src/test/java/com/thealgorithms/maths/AutoCorrelationTest.java (reformatted) @@ -34,4 +34,4 @@ } return result; } -}+} I don't really understand what I sould do here


Ok I fixed it

**Comment by AthinaSw** [comment]
> Please fix the PR check image

Done!

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by AthinaSw** [comment]
> Looks good, thanks!

You're welcome!

**Pull request merged**